ksqldb-engine/src/main/java/io/confluent/ksql/internal/TelemetryMetricsReporter.java

```diff
@@ -0,0 +1,141 @@
+import org.apache.kafka.common.metrics.MetricsContext;
+import org.apache.kafka.common.utils.AppInfoParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class TelemetryMetricsReporter implements MetricsReporter {
+
+  public static final String RESOURCE_LABEL_PREFIX = "resource.";
+  public static final String RESOURCE_LABEL_TYPE = RESOURCE_LABEL_PREFIX + "type";
+  public static final String RESOURCE_LABEL_VERSION = RESOURCE_LABEL_PREFIX + "version";
+  public static final String RESOURCE_LABEL_COMMIT_ID = RESOURCE_LABEL_PREFIX + "commit.id";
+  public static final String RESOURCE_CLUSTER_ID = RESOURCE_LABEL_PREFIX + "cluster.id";
+
+  private final Logger logger = LoggerFactory.getLogger(TelemetryMetricsReporter.class);
+  private final TelemetryReporter reporter;
+  private final Map<String, String> metricTags = new HashMap<>();
+  // ToDo: How is the cluster ID passed to this class
+  private final String clusterId = "";
+  private final String metricGroup;
+
+
+  public TelemetryMetricsReporter(
+          final String metricGroup,
+          final String providerNamespace,
+          final String resourceType,
+          final Map<String, ?> config
+  ) {
+    this(metricGroup, providerNamespace, resourceType, config, new TelemetryReporter());
+  }
+
+  protected TelemetryMetricsReporter(
+          final String metricGroup,
+          final String providerNamespace,
+          final String resourceType,
+          final Map<String, ?> config,
+          final TelemetryReporter reporter
+  ) {
+    this.metricGroup = metricGroup;
+    this.reporter = reporter;
+    configureTelemetryReporter(providerNamespace, resourceType, config, reporter);
+  }
+
+  @Override
+  public void report(final List<DataPoint> dataPoints) {
+    for (final DataPoint dataPoint : dataPoints) {
+      emitMetric(
+              dataPoint.getName(),
+              dataPoint.getTime(),
+              dataPoint.getValue(),
+              ImmutableMap.<String, String>builder()
+                      .putAll(metricTags).putAll(dataPoint.getTags()).build()
+      );
+    }
+  }
+
+  void emitMetric(
+          final String metricName,
+          final Instant timestamp,
+          final Object value,
+          final Map<String, String> metricTags
+  ) {
+    // Todo: Is it correct to use the KsqlProvider provided by ce-kafka?
+    final String fullMetricName = MetricsUtils.fullMetricName(
+            KsqlProvider.DOMAIN, metricGroup, metricName);
+    final MetricKey key = new MetricKey(fullMetricName, metricTags);
+
+    reporter.emitter().emitMetric(SinglePointMetric.gauge(key, (Double) value, timestamp));
+  }
+
+  private void configureTelemetryReporter(
+          final String providerNamespace,
+          final String resourceType,
+          final Map<String, ?> config,
+          final TelemetryReporter reporter
+  ) {
+    // Todo: Is it correct to use the KsqlProvider provided by ce-kafka?
```

Comment:
I may have asked this before, but I think this is from ce-metrics not ce-kafka right? Anyways I think that's correct: the only other related provider I'm aware is the KsqlHealthCheckProvider from cc-ksql-healthcheck which seems definitely not what we want.